### PR TITLE
Add more algorithm templates and template cleanup

### LINKED ISSUE #281: Add more algorithm templates
### Description

The algorithm template now works with grand-challenge, but it would be nice to demonstrate that outputting segmentations are equally simple to achieve.

Ideally, we would like to support three templates:
* [x] Segmentation (image -> image)
* [x] Detection (image -> multiple markers)
* [x] Classification (image -> classification label)
Furthermore, we would like to integrate these templates using the CLI
* [x] Nice way to generate this from the CLI
* [x] Make more generic template examples
* [x] Add test file injection for algorithms

### PR DESCRIPTION
**Pull request by silvandeleemput**
First #289 should be merged
Closes #281

* Adds Segmentation/Classification/Detection algorithm templates
* Adds test templating for expected outputs (changes to test.sh/test.bat and the pytests)
* Changed names for input files to generic input names lung->input_image

### CODE REVIEW

File: evalutils/templates/algorithm/cookiecutter.json [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
     "algorithm_name": "",
+    "algorithm_kind": [
+        "Classification",
+        "Segmentation",
+        "Detection"
```
Comment (jmsmkn) [inline evalutils/templates/algorithm/cookiecutter.json]: Do people have to type this out? Would be nice if they could use `C`, `S` or `D`. Maybe another issue?

File: evalutils/templates/algorithm/{{ cookiecutter.package_name }}/process.py.j2 [hunk 2]
```diff
@@ -1,74 +1,155 @@
...
-        self._scored_nodules = DataFrame()
 
     def process_case(self, *, idx, case):
-        lung_path = case["path"]
+        input_image_file_path = case["path"]
 
-        # Load the images for this case
-        lung = self._file_loaders["lung"].load_image(lung_path)
+        # Load the image for this case
+        input_image = self._file_loaders["input_image"].load_image(
```
Comment (jmsmkn) [inline evalutils/templates/algorithm/{{ cookiecutter.package_name }}/process.py.j2]: Haha, solved it already!

**Comment by jmsmkn** [comment]
Please solve the conflicts and see my comment above

File: evalutils/templates/algorithm/{{ cookiecutter.package_name }}/process.py.j2 (hunk 2 above)
Comment (silvandeleemput) [reply evalutils/templates/algorithm/{{ cookiecutter.package_name }}/process.py.j2]: Yep!

File: evalutils/templates/algorithm/cookiecutter.json (hunk 1 above)
Comment (silvandeleemput) [reply evalutils/templates/algorithm/cookiecutter.json]: Another issue might be better, this is in line with the Evaluations. I'll make one!

Comment (silvandeleemput) [reply evalutils/templates/algorithm/cookiecutter.json]: #291

File: evalutils/evalutils.py [hunk 3]
```diff
@@ -158,10 +180,158 @@ def process_case(self, *, idx: int, case: DataFrame) -> Dict:
         return {}
 
     def save(self):
-        with open(self._output_file, "w") as f:
+        with open(str(self._output_file), "w") as f:
             json.dump(self._case_results, f)
 
 
+class Algorithm(BaseAlgorithm):
```
Comment (jmsmkn) [inline evalutils/evalutils.py]: Is `BaseAlgorithm` used anywhere else but here? If not, why not combine the classes?

Comment (silvandeleemput) [reply evalutils/evalutils.py]: Yes, that would be much better. I'll do that.

**Pull request merged**